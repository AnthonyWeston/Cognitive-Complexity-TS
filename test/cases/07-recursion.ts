function f() {
    f();
}

function g() {
    g();
    g();
}

function h() {
    f();
}

class C {
    i() {
        this.i();
    }
}

const D = {
    j() {
        j();
    }
};
