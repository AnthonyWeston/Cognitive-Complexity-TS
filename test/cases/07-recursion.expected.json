{
    "07-recursion.ts": {
        "score": 9,
        "inner": [
            {
                "name": "f",
                "score": 1,
                "line": 1,
                "column": 1
            },
            {
                "name": "g",
                "score": 2,
                "line": 5,
                "column": 1
            },
            {
                "name": "h",
                "score": 0,
                "line": 10,
                "column": 1
            },
            {
                "name": "C",
                "score": 1,
                "line": 14,
                "column": 1,
                "inner": [
                    {
                        "name": "i",
                        "score": 1,
                        "line": 15,
                        "column": 5
                    }
                ]
            },
            {
                "name": "j",
                "score": 1,
                "line": 21,
                "column": 5
            },
            {
                "name": "l",
                "score": 1,
                "line": 26,
                "column": 11
            },
            {
                "name": "n",
                "score": 2,
                "line": 30,
                "column": 11
            },
            {
                "name": "p",
                "score": 1,
                "line": 35,
                "column": 11
            }
        ]
    }
}
